Makefile

```diff
@@ -1,5 +1,23 @@
 # We assume an active virtualenv for development
-install:
+include requirements.txt
+VENV_NAME ?= .venv
+VENV_ACTIVATE_FILE = $(VENV_NAME)/bin/activate
+VENV_ACTIVATE = . $(VENV_ACTIVATE_FILE)
+prereq: requirements.txt
+	pyenv install $(PY34)
+	pyenv install $(PY35)
+	pyenv install $(PY36)
+	pyenv install $(PY37)
+	pyenv global system $(PY34) $(PY35) $(PY36) $(PY37)
+	-@printf "\033[0;31mIMPORTANT\033[0m: please add \033[0;31meval \"\$$"
+	-@printf "(pyenv init -)\"\033[0m to your bash profile and restart your terminal before proceeding any further.\n"
+
+venv-create:
+	-@if [ ! -f $(VENV_ACTIVATE_FILE) ]; then python3 -mvenv ${VENV_NAME}; fi;	
+venv:
+	-@$(VENV_ACTIVATE)
+
+install: venv-create venv
```

Comment:
As discussed in person earlier, unfortunately depending on `venv` doesn't help because each line runs a different shell. We could use the Makefile changes I prepared here: https://github.com/elastic/rally/compare/master...dliappis:makefile_venv?expand=1 and add a prerequisite check for the presence of pyenv and finalized this PR.

There could be some pyenv specific setup issues on Ubuntu but on a few clean envs where I tested it things seemed to work alright.

Reply:
Modified as per our discussion to call . $(VENV_ACTIVATE_FILE) on every line where we execute python. Also added checks for pyenv in venv-create target.